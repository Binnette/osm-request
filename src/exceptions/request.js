export function RequestException(message) {
  this.message = message;
  this.name = 'RequestException';
}
