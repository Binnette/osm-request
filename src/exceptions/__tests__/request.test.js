import { RequestException } from '../request';

describe('GeoJSON exceptions', () => {
  describe('RequestException', () => {
    it('Should be well formatted', () => {
      const message = 'This is a test';
      const myFunction = () => {
        throw new RequestException(message);
      };

      expect(myFunction).toThrow(RequestException);
      expect(myFunction).toThrow(message);
      expect(myFunction).toThrowErrorMatchingSnapshot();
    });
  });
});
