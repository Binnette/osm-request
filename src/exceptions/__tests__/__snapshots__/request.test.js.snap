// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoJSON exceptions RequestException Should be well formatted 1`] = `"This is a test"`;
